Indicators/MovingAverageConvergenceDivergence.cs

```diff
@@ -95,8 +96,16 @@ protected override decimal ComputeNextValue(IndicatorDataPoint input)
 
             var macd = Fast.Current.Value - Slow.Current.Value;
 
-            Signal.Update(input.Time, macd);
-            Histogram.Update(input.Time, macd - Signal.Current.Value);
+            if (Fast.IsReady && Slow.IsReady)
+            {
+                Signal.Update(input.Time, macd);
+            }
+                
+            if (Signal.IsReady)
+            {
+                Histogram.Update(input.Time, macd - Signal.Current.Value);
```

Comment:
Believe this `if` could be moved into the one above. 
Instead of consuming the `IsReady` properties, we could directly use the returned value from the Update calls since we already have them